torch/fx/experimental/proxy_tensor.py

```diff
@@ -82,29 +92,35 @@ def unwrap_proxy(e):
         args[0].proxy = proxy_out
         proxy_out.node.meta['tensor_meta'] = _extract_tensor_metadata(args[0])
 
-    with no_dispatch():
-        real_out = func_overload(*args, **kwargs)
-
+    real_out = func_overload(*pytree.tree_map(unwrap_elem, args), **pytree.tree_map(unwrap_elem, kwargs))
+    # Seems like func isn't marked as inplace_view for some reason?
+    if torch.Tag.inplace_view in func_overload.tags:  # type: ignore[attr-defined]
+        with no_dispatch():
+            func_overload(*args, **kwargs)
+    pytree.tree_map(lambda x: check_metadata_consistency(x, ProxyTensor), (real_out, args, kwargs))
     return wrap_output(real_out, proxy_out)
 
+
 class ProxyTensor(torch.Tensor):
     proxy: fx.Proxy
+    elem: torch.Tensor
```

Comment:
would it be more accurate to make `elem` a `FakeTensor` here ? or Union of `torch.Tensor` and `FakeTensor`

Reply:
Well, `FakeTensor` is just a subclass of `torch.Tensor`, so I think `torch.Tensor` makes more sense here.